cmd/nerdctl/pull_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"gotest.tools/v3/assert"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+type cosignKeyPair struct {
+	publicKey  string
+	privateKey string
+	cleanup    func()
+}
+
+func newCosignKeyPair(t testing.TB) *cosignKeyPair {
+	td, err := os.MkdirTemp(t.TempDir(), "cosign-key-pair")
+	assert.NilError(t, err)
+
+	t.Setenv("COSIGN_PASSWORD", "1")
+
+	cmd := exec.Command("cosign", "generate-key-pair")
+	if out, err := cmd.CombinedOutput(); err != nil {
+		t.Fatalf("failed to run %v: %v (%q)", cmd.Args, err, string(out))
+	}
+
+	publicKey, err := filepath.Abs("cosign.pub")
+	if err != nil {
+		t.Fatalf("failed to get public key %v", err)
+	}
+
+	privateKey, err := filepath.Abs("cosign.key")
+	if err != nil {
+		t.Fatalf("failed to get private key %v", err)
+	}
+
+	return &cosignKeyPair{
+		publicKey:  publicKey,
+		privateKey: privateKey,
+		cleanup: func() {
+			_ = os.RemoveAll(td)
+		},
+	}
+}
+
+func TestImageVerifyWithCosign(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	keyPair := newCosignKeyPair(t)
+	defer keyPair.cleanup()
+	base := testutil.NewBase(t)
+	reg := newTestRegistry(base, "test-image-cosign")
+	defer reg.cleanup()
+	localhostIP := "127.0.0.1"
+	t.Logf("localhost IP=%q", localhostIP)
+	testImageRef := fmt.Sprintf("%s:%d/test-push-signed-image",
+		localhostIP, reg.listenPort)
+	t.Logf("testImageRef=%q", testImageRef)
+	base.Cmd("push", testImageRef, "--sign", "--cosign-key="+keyPair.publicKey).AssertOK()
+	base.Cmd("pull", testImageRef, "--verify", "--cosign-key="+keyPair.privateKey).AssertOK()
```

Comment:
Can we also assert that `--cosign-key=SOME-BAD-KEY` fails?

Reply:
yep, I did it.